src/appReview/slice.ts

```diff
@@ -0,0 +1,42 @@
+import { createSlice } from '@reduxjs/toolkit'
+import InAppReview from 'react-native-in-app-review'
+import { REHYDRATE, RehydrateAction } from 'redux-persist'
+import { getRehydratePayload } from 'src/redux/persist-helper'
+
+export interface State {
+  inAppRatingSupported: boolean
+  appRated: boolean
+  lastInteractionTimestamp: number | null
+}
+
+const initialState: State = {
+  // Android version >= 21 and iOS >= 10.3
+  // https://github.com/MinaSamir11/react-native-in-app-review#usage
+  inAppRatingSupported: !!InAppReview.isAvailable,
+  appRated: false,
+  lastInteractionTimestamp: null,
+}
+
+export const slice = createSlice({
+  name: 'appReview',
+  initialState,
+  reducers: {
+    inAppRatingSupported: (state, action) => {
+      state.inAppRatingSupported = action.payload
+    },
+    setAppRated: (state, action) => {
+      state.appRated = action.payload
+      state.lastInteractionTimestamp = Date.now()
+    },
+  },
+  extraReducers: (builder) => {
+    builder.addCase(REHYDRATE, (state, action: RehydrateAction) => ({
+      ...state,
+      ...getRehydratePayload(action, 'appReview'),
+    }))
+  },
```

Comment:
If there's no custom rehydration logic needed we can remove this. By default all reducers are rehydrated from persisted state at launch.

Reply:
Removed in a2d40f9bde8a242ba509db0333da7b149accad20!